src/pages/DisplayFixtures.jsx

```diff
@@ -122,14 +112,16 @@ const DisplayFixtures = () => {
           },
           body: JSON.stringify(update),
         });
-        setAllPredictions([...predictions, prediction]);
-        setEditMode(false);
         // console.log(`Predictions updated in database: ${JSON.stringify(predictions)}`)
-
       } catch (error) {
         console.error('Failed to create/update listing:', error);
       };
     };
+    const fetchedPredictions = await getPredictions();
+    setAllPredictions(fetchedPredictions);
+    setChangedPredictions([]);
+    setIsSaving(false);
+    setEditMode(false);
```

Comment:
after all changes are written & done, we can safely set loading state & edit mode to `false` :)